client/js/controllers/editor.js

```diff
@@ -1036,6 +1038,130 @@
+                            }
+                        }
+                        var curCmd = curTokens[0];
+                        if(!curCmd) return;
+                        var node_pos = nodenames.indexOf(curTokens[curTokens.length-2]);
+                        if (curTokens[curTokens.length-2]) {
+                            var brackets = curTokens[curTokens.length-2].split('(');
+                            if (brackets[1]) {
+                                var colon = brackets[1].split(':')[0];
+                                var colon_pos = nodenames.indexOf(colon);
+                            }
+                            var brack_pos = nodenames.indexOf(brackets[brackets.length-1].slice(0, -1));
+                        }
+                        if (valCmd) {
+                            var val_pos = membernodes.indexOf(valCmd);
+                        }
+                        if (node_pos > -1 || brack_pos > -1 || colon_pos > -1 || val_pos > -1) {
+                            var portsList = [];
+                            var dataTypeList = [];
+                            var values = members[val_pos];
+                            if (outports[node_pos]) {
+                                var ports = outports[node_pos];
+                            } else if (outports[brack_pos]) {
+                                var ports = outports[brack_pos];
+                            } else if (outports[colon_pos]) {
+                                var ports = outports[colon_pos];
+                            } else if (inports[node_pos]) {
+                                var ports = inports[node_pos];
+                            } else if (inports[brack_pos]) {
+                                var ports = inports[brack_pos];
+                            } else if (inports[colon_pos]) {
+                                var ports = inports[colon_pos];
+                            }
+                            if (values) {
+                                for (var i in values) {
+                                    portsList.push(values[i].name);
+                                    dataTypeList.push(values[i].data_type);
+                                }
+                                function getDataType(word) {
+                                    var index = portsList.indexOf(word);
+                                    return dataTypeList[index];
+                                }
+                                callback(null, portsList.map(function(word) {
+                                    return {
+                                        caption: word,
+                                        value: word,
+                                        meta: getDataType(word)
+                                    };
+                                }));
+                            } else if (ports) {
+                                for (var i in ports) {
```

Comment:
There is already a var i declared. You can create another one here or use the same but do not create another one